### ISSUE TITLE
Revenue analysis worksheet

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1684]
In GitLab by @kdjstudios on Apr 13, 2020, 08:21

**Submitted by:**  "rich.soltoff@answernet.com" <rich.soltoff@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/11129496

**Server:** Internal

**Client/Site:** NA

**Account:** NA

**Issue:** 

The problem is that if you add up the pieces, they do not add up to the total charges amount 

![SnipImage](/uploads/9301e5e7a4a1ebf36b9547ae998f2728/SnipImage.jpg)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 13, 2020, 08:58

@amishra @ranand @Robleski @tim.traylor 

It does not appear that the values are totaling correctly, as Rich mentioned. However, I may be using the wrong formula to check this. May someone please look into this and provide the formula used to validate the totals and if they are not adding up correctly please provide an explanation along with an estimate to fix.

I took the first row for example:

Individual Totals: 
* (58794.05+62681.43+2507+673.32-1218.51) = 123437.29

Totals comparison:
* 121677.30 - 123437.29 = -1759.99

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 15, 2020, 02:20

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 15, 2020, 05:29

@kdjstudios @tim.traylor 

Thank you for the update.

We looked into this and our findings are described below:

| Previous Recurring Total                                         | Previous Usage Total                                                                  | Previous Periodic Total                                        | Previous Charges Total                            | Previous Credits Total                                                                                 | Previous Late Charge Total                           |
|------------------------------------------------------------------|---------------------------------------------------------------------------------------|----------------------------------------------------------------|---------------------------------------------------|--------------------------------------------------------------------------------------------------------|------------------------------------------------------|
| This is the sum of all fee line items which have recurring true. | This is the sum of all transaction fee line items which upload from the usages file.  | This is the sum of all fee line item which has billing period. | This is the sum of all invoice new charges total. | This is the sum of all negative fee line item which have recurring false. | This is the sum of all late charges of each invoice.  |

* **Previous Charges Total:** calculate each invoice new charges amount. that's why when you will sum individual totals it calculates differently.

* Also New charges column is including the sum of all line items including the SD fee as well.

* But if you download the file as CSV and sum of the charges column you will get the same results as **Previous Charges Total**

Please let us know if we missed something.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 15, 2020, 08:49

@amishra 

* **Previous Charges Total:** calculate each invoice new charges amount. that's why when you will sum individual totals it calculates differently.
> That is why we are using this column to check the totals for the other columns correct? Or what columns may we sum together to check the "Charges Total" column.

* Also New charges column is including the sum of all line items including the SD fee as well.
> Which column is the "New Charges" you are referring to?

* But if you download the file as CSV and sum of the charges column you will get the same results as **Previous Charges Total**
> I am confused with what you are trying to state. Are you stating the CSV output does not match the UI and the PDF outputs?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 15, 2020, 22:50

@kdjstudios @tim.traylor 

Thank you for the update.



> > Which column is the "New Charges" you are referring to?

* **Previous Charges Total** column.



> I am confused with what you are trying to state. Are you stating the CSV output does not match the UI and the PDF outputs?

* No, I meant, sum of each previous charges column of each invoice balance is equal to the **Previous Charges Total.**

![image](/uploads/a8595c3d04ff5339cd183e7844607da1/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 16, 2020, 02:24

@kdjstudios @Robleski @tim.traylor 

**One of the reasons for this difference is because this report does not accommodate the positive staged fees/charges on the invoice.** There are multiple types of fees which play part in calculating the new charges of the invoice For eg, staged fees, SD Fees, taxes, etc.

* We took **Atlanta, AG, 03/27/2020 Revenue Analysis Worksheet**  as an example. 

Individual Totals:

* (22,578.47+6,294.17+0-552.37+149.60) = 28469.87

Totals comparison:

* 28469.87 - 29329.87 = -860

In the report, we found that there is one account with the same difference:

![image](/uploads/e94edab83f64dba79d148b6d56616bfd/image.png)

The invoice had staged fees with the same amount:

![image](/uploads/228b1d0ddcf2d838b3276363c0895ffa/image.png)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 16, 2020, 10:33

@ranand @tim.traylor @Robleski 

May someone please help me understand this? I feel like we have an issue with the calculation as @ranand describes above that fees are not being totaled correctly due to positive/negative, but this does not make any sense to me.

Maybe this will help too, as my understanding: The report needs to have the individual (Column) details add up into the report totals (Column). Also the report category totals need to add to match the report "Charges Totals".

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 17, 2020, 01:28

@kdjstudios @Robleski @tim.traylor 

As Ashish described the columns we have in the report in his above [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_57964).

Columns in the report:
* Previous Recurring Total: Sum of all the recurring Fees.
* Previous Usage Total: Sum of all the usages
* Previous Periodic Total: Sum of all billing period fees.
* Previous Charges Total: Sum of all new charges.
* Previous Credits Total: Sum of all credit fees.
* Previous Late Charge Total: Sum of all late fees.


As per the SAB:

`NEW CHARGES = recurring Fees + usages + billing period fees + late fees + SD fees + Staged Fees + taxes - credit fees - payments`

As you can see that apart from the Report columns, there are other types of Fees that contribute to the new charges of the invoice. 

**NOTE:** Previous Charges column displays the actual new charges of the invoice and the Previous Charges Total column sums the actual new charges of all the invoices in the report.

I hope it helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 17, 2020, 13:24

@tim.traylor @Robleski 

If I am understanding this correctly the New Charges column we are including other data/metrics that are not on this report (SD Fees, Staged Fees, Taxes, and Payments). If that is correct, I think we will want to review this with Gary to determine if this is correct or if we need to make any adjustments to the report.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 23, 2020, 09:34

In summary, the Misc column in the billing history page appears to not be calculated into this report. We believe that if adding it to the Periodic column should resolve the concern. I will be reaching out to Rich to review.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 24, 2020, 14:18

@amishra @ranand @Robleski @tim.traylor 

I was able to review this with Rich and he confirms that the Misc column charges from the Billing History Page should be in the periodic column on the Revenue Analysis report. He did want for us to confirm that we are not doubling the negative Misc charges with the credits column. Also He mentions the total column should not include payments, but on the comment here http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_58055 it states we do. So we need to double check that. 

In summary Rich needs for all of the columns in the revenue analysis report to add up to match the total column in the report. Then also the total columns should match the totals in the other reports (Billing cycle detailed, summary billing journal, etc.) Please post and estimate.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 29, 2020, 05:44

@kdjstudios @tim.traylor @Robleski 

Thank you for the update.

> I was able to review this with Rich and he confirms that the Misc column charges from the Billing History Page should be in the periodic column on the Revenue Analysis report. He did want for us to confirm that we are not doubling the negative Misc charges with the credits column. Also He mentions the total column should not include payments, but on the comment here http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_58055 it states we do. So we need to double check that.

* Misc column already including the credit charges on the Billing Cycle page.
* When we will same as the misc column on the **Periodic column** then the **credit section** charges double the report.

* The payment column does not including the column except the **Charges column** because they include the sum of each invoice line item.

> In summary Rich needs for all of the columns in the revenue analysis report to add up to match the total column in the report. Then also the total columns should match the totals in the other reports (Billing cycle detailed, summary billing journal, etc.) Please post and estimate.

*  As per my understanding, if we same as Misc **charges** with **Periodic column** they what do wee need to do for **Credits** section for the report.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 29, 2020, 10:01

@amishra @ranand @tim.traylor @Robleski 

You are completely contradicting what @ranand told us on the Thursday meeting when we commented http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_58258

Someone needs to explain exactly why the "total" columns do not equal the sum of the other columns.

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on Apr 29, 2020, 13:48

lets review on the sab call on 5/1 at 7am mt time. kyle are you able to join us? zoom is https://zoom.us/my/antechsvc1

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 30, 2020, 02:05

@kdjstudios @tim.traylor @Robleski 

As I stated on our Thursday call that there are various types of charges that get included in calculating the new charges and the Revenue analysis worksheet(RAW) does not show all of them.
And yes Rich is right that new charges do not include payments.

The example we discussed on Thursday is one of many and it was for positive staged fees from my above [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_58006).

Here are all the charges included in calculating Invoice New Charges.

| Charges added in New Charges of the Invoice  | Included in RAW | Header in RAW |
| ------ | ------ | ------ |
| Recurring Fees | Yes | Recurring Total |
| Usages | Yes | Usage Total | 
| Billing Period Fees | Yes | Periodic Total |
| late fees | Yes | Late Charge Total |
| SD fees | No | N/A |
| Staged fees | No | N/A |
| Taxes | No | N/A |
| Credit Fees/Negative staged fees | Yes | Credits Total |
| Holiday fees | No | N/A |

We need to figure out where to add these missing charges on the RAW report.

I hope this helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2020, 08:57

This was reviewed today. It was mentioned that @ranand knows what needs to be updated and included into the Periodic column of the RAW report to allow for the categories to sum together to equal the total column.

EDIT: Please post an estimate to resolve.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 7, 2020, 06:13

@kdjstudios  @tim.traylor @Robleski 

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Worked with Concepts, implementation ok|2.5|High Impact|2.5|5|

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 7, 2020, 09:29

Approving, as the report needs to be fixed.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 11, 2020, 02:54

@kdjstudios  @tim.traylor 

Thanks for the update.

While development we encounter a scenario that is as follows:-
For Account No
![image](/uploads/b51bc18a1c2791cdc746568ed87180cb/image.png)


![image](/uploads/0d489bee45900d448f41bac2a1a8a5a8/image.png)

*  There are two recurring fees added on account with the same **Additional fees** 
*  One has billing period **N/A** and second has billing period **A3**

*  On Revenue analysis report(RAW) the **Periodic Column** amount showing incorrectly because of both **additional ID's** are the same.

 So our query is how this needs to be handled? 

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 11, 2020, 08:26

@rranjan @Robleski @tim.traylor @amishra @ranand 

I am confused on what the problem is? From a quick look at the description, it would appear the one assigned to the A3 billing cycle is a "Yearly" fee, not a monthly one, and so it should only generate during that specified billing period. 

Or are you saying that even though the A3 fee should be generating only during that billing period. It is generating during all billing cycles/periods?

EDIT: Also why is this added to this ticket? this appears to be a completely separate from the RAW report? Or may someone please add more clarification and detail. Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on May 12, 2020, 06:25

@kdjstudios @Robleski 

* A3 gets generated only during the specified billing period. For billing and invoicing it is working correctly.

* The issue is only in the RAW report. Since **Paging Fee** and  **Yearly Pager Ins** are created on the same additional Fee ie **Pager Rental**, both the fees are coming under the periodic column.

* Although the total issue is fixed and in all other scenarios it is working fine. 

* In the given scenario, only the issue is **Pager Rental - Paging Fee** and **Pager Rental - Yearly Pager Ins** both are adding into the period column rather than one being in the **recurring column** and other in the **period column**.

NOTE: This is issue the of duplicate recurring fees. 

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 12, 2020, 11:40

@ranand @Robleski 

This is working correctly; to my knowledge. Those are fees setup in the "Recurring Fees" section on the account and should be in the "recurring" column; regardless if they are annual, quarterly, semi annual, monthly, weekly, etc; they are "recurring" just are recurring at different increments(Billing periods). Correct?

Just to confirm what fees are being entered into the "Periodic" column after this update. I would expect this should just be the Misc fees as pointed out in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_58006. Which if i am understanding correctly is just "one time / Manual" fees. Examples such as staged fees or additional fees on the draft invoice. Is that correct? Please confirm or explain in detail the differences.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on May 13, 2020, 05:59

@kdjstudios @Robleski 

* As you told us on our last Thursday call, all the missing fees will add to the periodic column.

* After this update all the missing fees listed in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_58541 will add to the periodic column. So the new calculation of periodic column will add the following fees.

| Fee | Description |
| ------ | ------ |
| Periodic Fee | the fee specific to billing period |
| SD Fee | the SD fee applied on the invoice |
| Positive/Debit Staged Fees | the Staged fee with positive value |
| Holiday Fee | the Holiday fee applied on the invoice |
| Taxes | the taxes applied on the invoice |

* The issue described by the http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1787#note_58888 is the existing issue with the report. I'll try to detail it out with the help of the RAW report. For the **Park Gardens Account** we have two recurring fees for the **Pager Rental**
  *  Pager Rental - Paging Fee | Monthly recurring Fee | **BaseRate: `$51`**
  *  Pager Rental - Yearly Pager Ins | Yearly Billing Fee | **BaseRate: `$25`**

![image](/uploads/8ab8ef434794f6f638fcfc5873843d10/image.png)

In the above snapshot both the fees are adding into **Recurring Column** and **Periodic Column**

**We have solved the issue of duplicate addition, but since both the fees are based on the same additional fee. Both the fees are now coming under the periodic column**

After this update, this record will look like the following

![image](/uploads/d3e0e807df5d2816bd22ad7d523db4c6/image.png)

**So, the remaining issue here is that since both the fees are based on the `Pager Rental`, the report is unable to differentiate between them that which is a recurring fee and which is a periodic fee. Hence, both the fees are coming under the periodic column**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 13, 2020, 10:58

@ranand @amishra  @Robleski 

It sounds like what we had previously discussed during the last meeting was not conveyed/described correctly. SO when i reviewed this with Rich it was not described to him correctly either. We will want to review this further with Rich I believe. My thoughts/notes below to review with Rich.

EDIT: I think we will only want to make updates/changes to the RAW report even though we may have found concerns with other reports during this review too.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 13, 2020, 11:44

To Review with Rich

Reports:

[summary_billing_journal_04_22_2020_Master.csv](/uploads/4d521c306c4c8bbd808abc1991ab17bc/summary_billing_journal_04_22_2020_Master.csv)

[revenue_analysis_worksheet_04_22_2020_Master.csv](/uploads/ebe237bf0aab448a805167eac6e0cfba/revenue_analysis_worksheet_04_22_2020_Master.csv)

[service_code_summary_04_22_2020_Master.csv](/uploads/7c2e6d107de7bc96463595d406a80fef/service_code_summary_04_22_2020_Master.csv)

[billing_cycle_detail_04_22_2020_Master.csv](/uploads/a722fc468ac437de4f14258a150ee6ed/billing_cycle_detail_04_22_2020_Master.csv)

### All reports and total buckets

Also I would image we want the "Totals" to match across at least these reports. Possibly others that I may not be thinking about. NOTE: Below I ran this for the El Paso site's 4/22/20 cycle and just to confirm which reports match their totals currently. 

* RAW 
  - $-4,310.97 in the credits. (This matches the BCD and SBJ)
  - $25,080.06 in the Usage. (This matches the SBJ)
  - $66,573.30 in the Recurring. (This matches the SBJ)
  - $1,395.44 in the Late Charge. (This matches the SBJ)
  - $00.00 in the Periodic.
  - $92,651.84 in the Total.
  - Once we find out where the above missing charges go, that should equal these totals out with the totals in the other reports.
  - Subtotals are missing $3,914.01 to match the total. This appears to be missing the MISC DR and Taxes from the SBJ report which is $3,914.01.

* SCS
  - $-2,959.96 in the credits. (This does not match credits in BCD, RAW, and SBJ reports. Why?)
  - $96,962.81 in the debits. (This appears to be including the "Taxes". Which then would match the BCD report. Possibly separate this out to better "match" other reports? Or update the header label to state "Debits & Taxes".)
  - $92,651.84 in the total.

* BCD 
  - -4,310.97 in the credits.
  - $95,611.80 in the debits.
  - $1,351.01 in the Taxes.
  - $92,651.84 in the total.

* SBJ 
  - $66,573.30 in the Monthly Charges. (This is the recurring in the RAW)
  - $25,080.06 in the Excess quantity. (This is the usage in the RAW)
  - $1,395.44 in the Late Charge. (This is the late charge in the RAW)
  - $2,563.00 in the Misc DR 
  - $-4,310.97  in the credits. (This is the late charge in the RAW and BCD)
  - $1,351.01 in the Taxes. (This is the taxes in the BCD)
  - Does not seem to have a total of current charges, so no need to match the total on this report? Or should we be adding a "Total New Charges" column. That total would be "$92,651.84‬" and it would match the other reports.

### Summary:

To fix the Raw report I believe we need to do two things:
1. Add the Taxes column. This will best match then both the SCS and SBJ report's "Tax" columns. Or we can add this to the "Recurring" column, but that may cause confusion as it then would not match the SBJ report's "Recurring" column. 
2. Add the "Misc DR" column from the SBJ report into the "Periodic" column. Possibly rename this column to "MISC DR" so it then better matches the SBJ report too. 
3. Optional: Would be to also update the SCS report to either breakdown the Taxes into it's own column, or update the header to better describe that they are included to avoid confusion.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 13, 2020, 13:40

@Robleski @amishra @ranand 

I was able to review with Rich and the following are the updates we would like to do and the order we would like to do them in.

**First:**

RAW:
* Add new "Taxes" column. This will match all other reports taxes columns
* Relabel "Periodic Charges" to "MISC DR". This column/calculation will need to match the SBJ report's "MISC DR".
* The "Recurring Column" and it column/calculation will need to match the SBJ report's "Monthly".

**Second:**

SCBS:
* Move taxes into their own new column before the Net column. 
* Remove taxes from debits/credits column. Will be included in the Net column

SBJ:
* Add new "Total Net Charges" column. This will match the "Totals/Net" columns from the other reports. 

NOTE: We may want to relabel some column headers to better match across the reports, but that will be part of the reporting project.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 15, 2020, 07:49

@kdjstudios  @Robleski 

Thanks for the update.

We have made changes in the Revenue Analysis Worksheet(RAW) report and setup branch on the dev server for your testing.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 15, 2020, 15:30

@rranjan 

This looks great. Lets now work on the two other reports. Hopefully we will have this ready for release by the next release.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 18, 2020, 05:55

@kdjstudios  @Robleski 

Thanks for the update.


## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|3.5|Worked with Concepts, implementation ok|4|High Impact|2.5|6.5|

#TestCases:

**SCBS:**
1. Taxes will have its own column and taxes will be removed from credit and debit columns
1. Total Net =  Total Debit + Total Credit + Total Taxes
1. Similar changes its PDF and CSV

**SBJ:**
1. Add New column "Net Charges", it will be same as RAW's Charges column
1. Similar changes its PDF and CSV

**NOTE:** Total Net will not match in case of SD line items in the billing cycle invoices and as discussed this case will be handled in separate ticket

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 18, 2020, 07:41

@kdjstudios  @Robleski 

We have made changes to other reports and setup branch on the dev server for your testing.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @ranand on May 18, 2020, 08:43

@kdjstudios @Robleski 

Please ignore the above comment of @rranjan. 

In **`Summary billing journal CSV`** there are already two columns for new charges/Net Charges. 
1. **New Charges:** Will it be equal to the "Net Charges" of the report, because currently it is not or do we need to add a new column for the invoice new charges in CSV.
1. **AR_new_charges:** Do we need to change it?

NOTE: These columns are only in the SBJ CSV. Please review it's CSV and let us know what all changes we need to do in its CSV.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 18, 2020, 11:17

@ranand 

What is the current calculation of the "new charges" column on the CSV?

What is the current calculation of the "AR_new Charges" column on the CSV?  

If neither of them match the calculation for the "Net Charges" that we are requesting above, then they are different metrics and we will need to add "Net Charges" as a new column. If one or both of the above are the exact same calculation of the "net charges" then we may be able to just relabel the Header on the CSV to match. We still need to add the column to the PDF and web UI version by the sounds of your response, is that correct?

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 19, 2020, 08:24

@kdjstudios  @Robleski 

Thanks for the update.


*  new charges and AR_new_charges both calculate the invoice amount after applying the invoice credit fees to the previous invoice.

> If neither of them match the calculation for the "Net Charges" that we are requesting above, then they are different metrics and we will need to add "Net Charges" as a new column. If one or both of the above are the exact same calculation of the "net charges" then we may be able to just relabel the Header on the CSV to match. We still need to add the column to the PDF and web UI version by the sounds of your response, is that correct?

*  Neither of them is matching to invoice new charges. Both of them represent different values. So we have added a new column for the invoice new charges.


We have set up a branch on the dev server for your testing.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 19, 2020, 13:41

@ranand @amishra @rranjan @Robleski 

I was able to run the following reports and here are what I found.

**SBJ**
- CSV: [summary_billing_journal_03_27_2020_Master.xlsx](/uploads/465c0d300bd88cfbc07f8aba43892935/summary_billing_journal_03_27_2020_Master.xlsx)
  - Missing the Billing Cycle in the top.
  - What are all the "N/A" and "NIL" metrics? Should we be removing these?
  - What is the "Resource ID Total" Metric? 
- WEB & PDF
  - Looks okay.

**RAW**
- CSV: [revenue_analysis_worksheet_03_27_2020_Master.xlsx](/uploads/90b914bfd50b301c57c065bf7174cfd3/revenue_analysis_worksheet_03_27_2020_Master.xlsx)
  - Headers are misaligned. 
  - Also what are the Comments/Explanations? I do not see that on the UI or PDF.
  - Missing the Billing Cycle in the top.
- WEB: 
  - Looks okay.
- PDF: [revenue_analysis_worksheet_03_27_2020_Master.pdf](/uploads/34a24620d89b131cf718dd546c35c638/revenue_analysis_worksheet_03_27_2020_Master.pdf)
  - Half of the report is missing, due to horizontal width of the page I presume?


**SCBS**
- CSV: [service_code_summary_03_27_2020_Master.csv](/uploads/386e3399e8c4af99c03d6f898258b9b2/service_code_summary_03_27_2020_Master.csv)
  - Missing the billing cycle in the top. 
- WEB & PDF
  - Looks okay.

**BCD**
- CSV: [billing_cycle_detail_03_27_2020_Master.csv](/uploads/dfb6c00808420f9688a316f142d90f2f/billing_cycle_detail_03_27_2020_Master.csv)
  - Missing the billing cycle in the top. 
- WEB & PDF
  - Looks okay.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 20, 2020, 07:07

@kdjstudios  @Robleski 

Thanks for the update.

**SBJ**

CSV:
>Missing the Billing Cycle in the top.

We have added it

>What are all the "N/A" and "NIL" metrics? Should we be removing these?
 
*  Not Applicable(N/A)/NIL is showing for the metrics whose total sum is not possible. i.e- date,name etc.

> What is the "Resource ID Total" Metric?

*  It was basically showing some garbage value for the resource ids.

*  We have added N/A for the resource ID Total.

* Please let us know whether we need to keep N/A and NIL or we need to remove them from the CSV?

**RAW**

CSV:
> Headers are misaligned.

We have fixed it.

>Also what are the Comments/Explanations? I do not see that on the UI or PDF.

It is just a header with an empty value.

>Missing the Billing Cycle in the top.

We have added it.

PDF:

> Half of the report is missing, due to the horizontal width of the page I presume?

This was an existing issue even if we change the pdf orientation to **landscape** some fields will still be not visible because the report has large number of columns.

**SCBS**

CSV:

> Missing the billing cycle in the top.

We have added it.

**BCD**

CSV:

> Missing the billing cycle in the top.

We have added it.


We have made the changes and setup branch on the dev server for your testing.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 20, 2020, 09:38

@rranjan 

**SBJ**

CSV:


> What are all the "N/A" and "NIL" metrics? Should we be removing these?

> * Not Applicable(N/A)/NIL is showing for the metrics whose total sum is not possible. i.e- date,name etc.

> What is the "Resource ID Total" Metric?

> * It was basically showing some garbage value for the resource ids.

Please remove all the of the above values that can not be calculated. No reason to include them on the report.


**RAW**

CSV:

> Also what are the Comments/Explanations? I do not see that on the UI or PDF.

> * It is just a header with an empty value.

Please remove it.

PDF:

> Half of the report is missing, due to the horizontal width of the page I presume?

> * This was an existing issue even if we change the pdf orientation to **landscape** some fields will still be not visible because the report has large number of columns.

@Robleski What should we be doing here to insure all of the report fits on the PDF? I understand it was an existing issue, but I do not like the idea of not fixing this also while we are fixing the report.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 21, 2020, 01:12

@kdjstudios  @Robleski 

Thanks for the update.

We have removed all the values requested for CSV and setup branch on the dev server for your testing.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 21, 2020, 13:11

Looks good to me moving to be released this weekend.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 23, 2020, 05:05

@kdjstudios @Robleski 

We have successfully released on both of the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 23, 2020, 15:06

I wrote:

> The updates to the Revenue Analysis Worksheet, Summary Billing Journal, and Service Code Breakdown reports have been completed and are now in production. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 28, 2020, 11:33

closed
